kafka/retry_lazy_client.go

```diff
@@ -0,0 +1,346 @@
+package kafka
```

Comment:
Allow recovery from transient errors such as connection refused during a broker restart

RetryLazyClient wraps the LazyClient with retries and timeouts similar to how LazyClient wraps the Client with lazy instantiation

Uses the Terraform Plugin SDKv2 retry mechanism
https://www.terraform.io/plugin/sdkv2/resources/retries-and-customizable-timeouts
https://pkg.go.dev/github.com/hashicorp/terraform-plugin-sdk/v2/helper/resource